airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -725,19 +727,36 @@ def generate_scd_type_2_model(self, from_table: str, column_names: Dict[str, Tup
         {{ field }},
       {%- endfor %}
       {{ cursor_field }} as {{ airbyte_start_at }},
-      {{ lag_begin }}({{ cursor_field }}) over (
```

Comment:
I think most of this doesn't need to be conditional on `destination`, since `lag_begin` and `lag_end` are being set for all destinations anyway. This feels a bit more clear to me (since it's easier to see the clickhouse-specific changes):

```
      {{ lag_begin }}({{ cursor_field }}) over (
        partition by {{ primary_key_partition | join(", ") }}
        order by
            {{ cursor_field }} {{ order_null }},
            {{ cursor_field }} desc,
            {{ col_emitted_at }} desc{{ cdc_updated_at_order }}
        {{ lag_end }}
      ) as {{ airbyte_end_at }},
      {{ '{%- if var("destination") == "clickhouse" %}' }}
        case when _airbyte_active_row_num = 1{{ cdc_active_row }} then 1 else 0 end as {{ active_row }},
      {{ '{%- else %}' }}
        case when row_number() over (
          partition by {{ primary_key_partition | join(", ") }}
          order by
              {{ cursor_field }} {{ order_null }},
              {{ cursor_field }} desc,
              {{ col_emitted_at }} desc{{ cdc_updated_at_order }}
        ) = 1{{ cdc_active_row }} then 1 else 0 end as {{ active_row }},
      {{ '{%- endif %}' }}
```

arguably even the `= 1{{ cdc_active_row }} then 1 else 0 end as {{ active_row }},` could be extracted, but that felt a bit messier

Reply:
I made this to match the version before the Clickhouse PR got merged. Also the `_airbyte_active_row_num` is conditional to Clickhouse destination.

Reply:
tl;dr I think both versions work correctly, but feel free to merge as-is

I'm running both versions on my laptop and (so far) they've produced identical (except whitespace) output for snowflake. Haven't actually been able to run clickhouse locally, but I expect it would also work correctly with both versions. That said, I don't feel super strongly about this; definitely don't consider this a blocker.

for clickhouse, the refactored version will produce
```sql
      {{ lag_begin }}({{ cursor_field }}) over (
        partition by {{ primary_key_partition | join(", ") }}
        order by
            {{ cursor_field }} {{ order_null }},
            {{ cursor_field }} desc,
            {{ col_emitted_at }} desc{{ cdc_updated_at_order }}
        {{ lag_end }}
      ) as {{ airbyte_end_at }},
      case when _airbyte_active_row_num = 1{{ cdc_active_row }} then 1 else 0 end as {{ active_row }},
```
(tweaked whitespace for clarity)

and for other destinations:
```sql
      {{ lag_begin }}({{ cursor_field }}) over (
        partition by {{ primary_key_partition | join(", ") }}
        order by
            {{ cursor_field }} {{ order_null }},
            {{ cursor_field }} desc,
            {{ col_emitted_at }} desc{{ cdc_updated_at_order }}
        {{ lag_end }}
      ) as {{ airbyte_end_at }},
      case when row_number() over (
        partition by {{ primary_key_partition | join(", ") }}
        order by
            {{ cursor_field }} {{ order_null }},
            {{ cursor_field }} desc,
            {{ col_emitted_at }} desc{{ cdc_updated_at_order }}
      ) = 1{{ cdc_active_row }} then 1 else 0 end as {{ active_row }},
```

I.e. the `airbyte_end_at` column is defined identically (since `lag_begin="lag"` and `lag_end=""`), but the `active_row` column is different (as you pointed out, clickhouse uses `_airbyte_active_row_num`, other destinations use the `case when row_number()` thing).